ttnn/cpp/ttnn/operations/conv/conv_transpose2d/conv_transpose2d.cpp

```diff
@@ -639,14 +593,121 @@ ConvT2DSliceAttr::ConvT2DSliceAttr(
+    const tt::tt_metal::DataType conv_input_dtype = (input_dtype == tt::tt_metal::DataType::FLOAT32)
+                                                        ? tt::tt_metal::DataType::FLOAT32
+                                                        : tt::tt_metal::DataType::BFLOAT16;
+    const uint32_t input_datum_size = conv_input_dtype == tt::tt_metal::DataType::FLOAT32 ? 4 : 2;
+    uint32_t input_size = shard_shape[0] * shard_shape[1] * input_datum_size;
+
+    sliding_window::ParallelConfig parallel_config = {
+        .grid = sliced_input_tensor_memory_config.shard_spec().value().grid,
+        .shard_scheme = sliced_input_tensor_memory_config.memory_layout(),
+        .shard_orientation = sliced_input_tensor_memory_config.shard_spec().value().orientation};
+    // Create SlidingWindowConfig for precise halo calculation
+    sliding_window::SlidingWindowConfig slice_halo_config;
+    slice_halo_config.batch_size = batch_size;
+    slice_halo_config.input_hw = {input_slice_height, input_slice_width};
+    slice_halo_config.window_hw = {kernel_size[0], kernel_size[1]};
+    slice_halo_config.stride_hw = {stride[0], stride[1]};
+    slice_halo_config.padding = this_slice_padding;
+    slice_halo_config.output_pad_hw = {this_output_padding.at(0), this_output_padding.at(1)};
+    slice_halo_config.dilation_hw = {dilation[0], dilation[1]};
+    slice_halo_config.num_cores_nhw = get_num_cores_channels_from_parallel_config(parallel_config);
+    slice_halo_config.core_range_set = sliced_input_tensor_memory_config.shard_spec().value().grid;
+    slice_halo_config.snap_to_tile = true;
+    const uint32_t input_channels_alignment = get_input_channels_alignment(
+        conv_config.shard_layout.value(),
+        conv_config.output_layout,
+        slice_config.num_slices > 1,
+        false,  // Matmul based convs should never take the slicing route.
+        std::nullopt);
+    uint32_t precise_max_halo_bytes =
+        sliding_window::calculate_precise_halo_output_elems(slice_halo_config, shard_shape) * input_datum_size;
+    auto compute_grid = device->compute_with_storage_grid_size();
+
+    ShardOrientation shard_orientation =
+        conv_config.transpose_shards ? ShardOrientation::COL_MAJOR : ShardOrientation::ROW_MAJOR;
+
+    sliding_window::ParallelConfig output_parallel_config =
+        determine_output_parallel_config(parallel_config, compute_grid, output_channels, shard_orientation, false);
+
+    uint32_t padded_in_channels = tt::round_up(
+        input_channels, tt::constants::TILE_WIDTH * get_num_cores_channels_from_parallel_config(parallel_config));
+    uint32_t padded_out_channels = tt::round_up(
+        output_channels,
+        tt::constants::TILE_WIDTH * get_num_cores_channels_from_parallel_config(output_parallel_config));
+    ttnn::Shape folded_weights_shape({1, 1, padded_in_channels * kernel_size[0] * kernel_size[1], padded_out_channels});
```

Comment:
this is approx size of weights tensor, this is fine for auto sharding will this be a problem for slicing?
Should we add function which just returns shape of the prepared weights tensor?